src/Renci.SshNet/Security/Cryptography/Ciphers/AEADCipher.cs

```diff
@@ -0,0 +1,423 @@
+            : base(key, blockSize, mode, padding)
+        {
+            _blockSize = blockSize;
+            _padding = padding;
+
+            // Get the IV length of bytes for AES-GCM (12 bytes) from initial IV.
+            // Length sz is basd on [RFC5647] Section 7.1 .Auguest 2009
+            _iv = new byte[NONCE_BYTES];
+            Array.Copy(iv, 0, _iv, 0, NONCE_BYTES);
+
+            _key = new byte[key.Length];
+            _key = key;
+
+            // Key Size Check for AES-GCM
+            // AES128-GCM: 16 bytes [RFC5116] Section 5.2 .January 2008
+            // AES192-GCM: 24 bytes [RFC5084] Section 1.3 .November 2007
+            // AES256-GCM: 32 bytes [RFC5116] Section 5.3 .January 2008
+            if (!(_key.Length == 16 || _key.Length == 24 || _key.Length == 32))
+                throw new ArgumentException(string.Format("KeySize '{0}' is not valid for this algorithm.", _key.Length));
+
+            cipher = new AesGcm(_key);
+
+            // If cipher is null throw exception
+            if (cipher == null)
+                throw new ArgumentException(string.Format("Invalid AES-GCM Class Cipher."));
+        }
+#if DEBUG_AESGCM
+
+        /// <summary>
+        /// GetHexStringFrom.
+        /// </summary>
+        /// <returns>String.</returns>
+        public static string GetHexStringFrom(byte[] byteArray)
+        {
+            return  BitConverter.ToString(byteArray); //To convert the whole array
+        }
+
+#endif
+
+        /// <summary>
+        /// AEAD AES-GCM Encrypt Function
+        /// </summary>
+        /// <param name="inputBuffer">The input data which contains packet length field (Additional Authenticated Data)
+        /// and the plain text.
+        /// </param>
+        /// <returns>
+        /// The number of bytes encrypted.
+        /// </returns>
+        // Structure of the inputBuffer:
+        // |------4 bytes------||-------------------Plain Text--------------------|
+        // [packet length field][padding length field sz][payload][random paddings]
+        private byte[] GcmEncrypt(byte[] inputBuffer)
+        {
+            if (inputBuffer.Length - PACKET_LENGTH_FIELD_SZ < _blockSize)
+                throw new ArgumentException("Invalid input buffer");
+
+            var tag = new byte[TAG_SIZE];
+            var cipherText = new byte[inputBuffer.Length - PACKET_LENGTH_FIELD_SZ];
+
+            var associatedData = new ReadOnlySpan<byte>(inputBuffer, 0, PACKET_LENGTH_FIELD_SZ);
+            var plainText = new ReadOnlySpan<byte>(inputBuffer, PACKET_LENGTH_FIELD_SZ, cipherText.Length);
+
+#if DEBUG_AESGCM
+
+            string var_iv = GetHexStringFrom(_iv);
+            string var_key = GetHexStringFrom(_key);
+            string var_pt = GetHexStringFrom(plainText.ToArray());
+            string var_aad = GetHexStringFrom(associatedData.ToArray());
+
+            Console.WriteLine("[DEBUG] In GcmEncrypt()");
```

Comment:
Have you considered other possibilities for outputting debug information? ```ILogger``` is not viable yet, but I would prefer ```System.Diagnostics.Debug.WriteLine``` over ```Console.WriteLine```.

Reply:
yes, I will follow your suggestion as at the time I wrote the code I wasn't too familiar with the C# DEBUG utility. Thank you for the feedback.

Reply:
I am getting this error 
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0246	The type or namespace name 'CngKey' could not be found (are you missing a using directive or an 
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0246	The type or namespace name 'CngAlgorithm' could not be found (are you missing a using directive or an assembly reference?)

Is because these are out of date in netstandard2.0 and netstandard2.1 ?